Client/Application/AppDelegate.swift

```diff
@@ -189,10 +190,88 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         // Run an invalidate when we come back into the app.
         profile.panelDataObservers.activityStream.invalidate(highlights: true)
 
+        setUpDeepLinks(application: application)
+
         log.debug("Done with setting up the application.")
         return true
     }
 
+    func setUpDeepLinks(application: UIApplication) {
```

Comment:
Unfortunately it relies on a lot of vars already in AppDelegate to push the controllers, assign certain delegates to VCs so it's convenient to just leave it there, I can make a bug to place it somewhere else if we really want to.